nvflare/fuel/f3/cellnet/cell.py

```diff
@@ -0,0 +1,1757 @@
+from nvflare.fuel.f3.comm_config import CommConfigurator
+from nvflare.fuel.f3.communicator import Communicator, MessageReceiver
+from nvflare.fuel.f3.connection import Connection
+from nvflare.fuel.f3.drivers.driver_params import DriverParams
+from nvflare.fuel.f3.endpoint import Endpoint, EndpointMonitor, EndpointState
+from nvflare.fuel.f3.message import Message
+from nvflare.fuel.f3.stats_pool import StatsPoolManager
+
+_BULK_CHANNEL = "cellnet.bulk"
+_ONE_MB = 1024 * 1024
+
+
+class TargetMessage:
+    def __init__(
+        self,
+        target: str,
+        channel: str,
+        topic: str,
+        message: Message,
+    ):
+        self.target = target
+        self.channel = channel
+        self.topic = topic
+        self.message = message
+
+    def to_dict(self):
+        return {
+            "target": self.target,
+            "channel": self.channel,
+            "topic": self.topic,
+            "message": {"headers": dict(self.message.headers), "payload": self.message.payload},
+        }
+
+    @staticmethod
+    def from_dict(d: dict):
+        msg_dict = d.get("message")
+        msg = new_message(headers=msg_dict.get("headers"), payload=msg_dict.get("payload"))
+        return TargetMessage(target=d.get("target"), channel=d.get("channel"), topic=d.get("topic"), message=msg)
+
+
+class CellAgent:
+    """
+    A CellAgent represents a cell in another cell.
+    """
+
+    def __init__(self, fqcn: str, endpoint: Endpoint):
+        """
+
+        Args:
+            fqcn: FQCN of the cell represented
+        """
+        err = FQCN.validate(fqcn)
+        if err:
+            raise ValueError(f"Invalid FQCN '{fqcn}': {err}")
+
+        self.info = FqcnInfo(FQCN.normalize(fqcn))
+        self.endpoint = endpoint
+
+    def get_fqcn(self):
+        return self.info.fqcn
+
+
+class _CB:
+    def __init__(self, cb, args, kwargs):
+        self.cb = cb
+        self.args = args
+        self.kwargs = kwargs
+
+
+class _Registry:
+    def __init__(self):
+        self.reg = {}  # channel/topic => _CB
+
+    @staticmethod
+    def _item_key(channel: str, topic: str) -> str:
+        return f"{channel}:{topic}"
+
+    def set(self, channel: str, topic: str, item):
+        key = self._item_key(channel, topic)
+        self.reg[key] = item
+
+    def append(self, channel: str, topic: str, item):
+        key = self._item_key(channel, topic)
+        item_list = self.reg.get(key)
+        if not item_list:
+            item_list = []
+            self.reg[key] = item_list
+        item_list.append(item)
+
+    def find(self, channel: str, topic: str):
+        item = self.reg.get(self._item_key(channel, topic))
+        if not item:
+            # try topic * in channel
+            item = self.reg.get(self._item_key(channel, "*"))
+
+        if not item:
+            # try topic * in channel *
+            item = self.reg.get(self._item_key("*", "*"))
```

Comment:
This use the same condition. Needs to place in the same condition block.

Reply:
The item value changes at line 134 so it changes the logic if the two blocks are combined.